tools/check_invalid_syntax

```diff
@@ -1,11 +1,37 @@
 #!/bin/sh
+
+exit_code=0
+GIT_CMD="git --no-pager grep --break  --color -n -P"
+
 check_loadtest() {
-    CHECK_LOADTEST_RESULT=$(git --no-pager grep --break -P "\sloadtest\s?\(?(?:\"|')?[\w\/]+\.pm(?:\"|'),?.*\)?")
+    CHECK_LOADTEST_RESULT=$(${GIT_CMD} "\sloadtest\s?\(?(?:\"|')?[\w\/]+\.pm(?:\"|'),?.*\)?")
     if [ ! -z "$CHECK_LOADTEST_RESULT" ]; then
-        echo $CHECK_LOADTEST_RESULT
+        echo "${CHECK_LOADTEST_RESULT}"
         echo "Invalid syntax found for loadtest call. '.pm' extensions are not allowed at test distribution level"
-        return $?
+        exit_code=$(($exit_code+1))
+    fi
+    echo "Finished checking loadtests"
+}
+
+check_versions(){
+    CHECK_IS_LEAP_RESULT=$(${GIT_CMD} "is_leap[\(\s](\"|')\d{2}(?!>\.\d)\1")
```

Comment:
Please help me understand:
 - is_leap sub has `\d{2,}\.\d` regex for checking version format
 - it's pushed to version_utils query `/^(?(?!.*\+$)(?<op>[<>=]|[<>]=))(?<qv>$regex)(?<plus>\+)?$/i` as $regex

What is `>` in `(?!>\.\d)` checking for?

Reply:
@kravciak it's kinda an annoying check tbh xD but I'm rather checking for a negative match.

Reply:
I think that proves the need of add a bash comment to explain that the intention of the regexp is to find a specific wrong use.